File: c_files/raylib/vampire_survivor_clone/collision_handler.c
#include "collision_handler.h"
#include "enemy_list.h"
#include "player.h"
#include "bullet_list.h"
#include "enemy.h"
#include "bullet.h"

void playerCollisionEnemy(EnemyList *enemy_list, Player *player)
{
    for (int i = 0; i < enemy_list->size; i++)
    {
        if (CheckCollisionRecs((Rectangle){player->x,
                                           player->y,
                                           player->width,
                                           player->health},
                               (Rectangle){enemy_list->list[i].x,
                                           enemy_list->list[i].y,
                                           enemy_list->list[i].width,
                                           enemy_list->list[i].height}))
        {
            player->health -= enemy_list->list[i].damage;
        }
    }
}

File: python_files/raylib/simple_space_survival/collision_handler.py
import pyray as p

from player import Player
from enemy_list import EnemyList
from enemy_list import Enemy


class Collision_handler:
    def __init__(self, player: Player, enemy_list: EnemyList) -> None:
        self.player: Player = player
        self.enemy_list: EnemyList = enemy_list

    def enemy_collision_bullet(self) -> None:
        dead_enemy_list: list[Enemy] = []
        for enemy in self.enemy_list.list:
            enemy.tint = p.WHITE # default enemy tint
            for bullet in self.player.bullet_list.list:
                if p.check_collision_recs(enemy.dest, bullet.dest):
                    enemy.tint = p.RED # change tint to red per collision
                    self.player.attack(enemy)
                    if not enemy.is_alive():
                        dead_enemy_list.append(enemy)

        for dead_enemy in dead_enemy_list:
            if dead_enemy in self.enemy_list.list:
                self.enemy_list.list.remove(dead_enemy)
